lib/infrastructure/persistence/pity_counter_data_service.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:hive/hive.dart';
+import 'package:shiori/domain/services/persistence/pity_counter_data_service.dart';
+
+class PityCounterDataServiceImpl implements PityCounterDataService {
+  late Box<int> _pityCounterBox;
```

Comment:
My idea here is to store a separate counter for `standardBanner`, `limitedBanner1` and `limitedBanner2` and `weaponsBanner`